File: test/fixtures/java-maven/src/main/java/com/example/app/service/OrderService.java
package com.example.app.service;

import com.example.domain.order.Order;
import com.example.domain.order.OrderItem;
import com.example.infrastructure.repository.OrderRepository;

public class OrderService {
    private OrderRepository repository;

    public OrderService(OrderRepository repository) {
        this.repository = repository;
    }

    public void createOrder(Order order) {
        repository.save(order);
    }

    public Order getOrder(String id) {
        return repository.findById(id);
    }
}

File: test/fixtures/swift-project/Sources/Application/OrderService.swift
import Foundation
import Domain
import Infrastructure
@_exported import Domain

/// Order service (application layer) — orchestrates domain and infrastructure
public class OrderService {
    private let repository: OrderRepository

    public init(repository: OrderRepository) {
        self.repository = repository
    }

    public func createOrder(id: String, items: [OrderItem]) -> Order {
        let order = Order(id: id, items: items)
        repository.save(order)
        return order
    }

    public func getOrder(id: String) -> Order? {
        return repository.findById(id)
    }
}
